src/services/apiService.ts

```diff
@@ -476,11 +465,12 @@ export class ApiService {
         return page;
     }
 
-    public async getApiHostnames(apiName: string): Promise<string[]> {
-        const query = `apis/${apiName}/hostnames`;
-        const pageOfHostnames = await this.mapiClient.get<Page<Hostname>>(query, [await this.mapiClient.getPortalHeader("getApiHostnames")]);
-        const hostnameValues = pageOfHostnames.value.map(x => x.properties.value);
+    public async getHostnames(): Promise<string[]> {
+        const query = `gateway/hostnames`;
+        const result = await this.mapiClient.get<{
```

Comment:
Please always prefer specific types (you have `Hostname` defined above) instead of inline types like `{ hostnames: string[] }`, for better code self-documenting. This, of course, makes sense only if you add jsdocs comments as well.

Reply:
added!